shell/platform/fuchsia/flutter/vsync_recorder.cc

```diff
@@ -12,9 +12,9 @@ namespace {
 
 std::mutex g_mutex;
 
-// Since we don't have any presentation info until we call |Present| for the
-// first time, assume a 60hz refresh rate in the meantime.
-constexpr fml::TimeDelta kDefaultPresentationInterval =
+// Assume a 60hz refresh rate before we have enough past
+// |fuchsia::scenic::scheduling::PresentationInfo|s to calculate it ourselves.
+static constexpr fml::TimeDelta kDefaultPresentationInterval =
```

Comment:
Can this use `VsyncWaiter::GetDisplayRefreshRate` to compute the interval? Android and iOS implement it in `shell/platform/android/vsync_waiter_android.cc` and `shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm` respectively. Fuchsia probably should also implement it in `shell/platform/fuchsia/flutter/vsync_waiter.cc`.

Reply:
This requires a bigger change, as Flutter now needs to compute it itself. To do this safely, we'd need to use `RequestPresentationTimes` to ask for information for the next >=2 vsyncs, take that difference and report it. However we'd also need to keep our list updated as we get later and more accurate information (that may change compared to what we already have).

I was going to add all this as a separate PR, as we would use this new information also to change how we schedule frames too and so it all goes together. If that sounds good to you, then I'd like to just do that and keep this PR as is.

Reply:
Sure, adding this in another PR sounds good. Let's keep this PR simple :)